qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -217,11 +207,29 @@ def status(self) -> OptimizationResultStatus:
         """
         return self._status
 
-    @status.setter  # type: ignore
-    def status(self, status: OptimizationResultStatus) -> None:
-        """Set a new termination status.
+    @property
+    def variables(self) -> List[Variable]:
+        """Returns the list of variables of the optimization problem.
 
-        Args:
-            status: The new termination status.
+        Returns:
+            The list of variables.
         """
-        self._status = status
+        return self._variables
+
+    @property
+    def variables_dict(self) -> Dict[str, int]:
+        """Returns the optimal value as a dictionary of the variable name and corresponding value.
```

Comment:
Likely `variables_dict` should return `Dict[str, float]` ?

Reply:
Yes, you are right.